articles/tests/test_views.py

```diff
@@ -262,3 +264,307 @@ def test_get_article_with_invalid_slug_fail(self) -> None:
+        return {"HTTP_AUTHORIZATION": f"Bearer {token}"}
+
+    def test_create_comment(
+        self,
+    ) -> None:
+        response = self.client.post(
+            reverse("comments", kwargs={"slug": self.article.slug}),
+            data=self.data,
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(str(self.data.get("body")), self.data.get("body"))
+
+    def test_create_comment_without_authentication(
+        self,
+    ) -> None:
+        response = self.client.post(
+            reverse("comments", kwargs={"slug": self.article.slug}),
+            data=self.data,
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertEqual(
+            json.loads(response.content).get("detail"),
+            "Authentication credentials were not provided.",
+        )
+
+    def test_get_all_comments(
+        self,
+    ) -> None:
+        response = self.client.get(
+            reverse("comments", kwargs={"slug": self.article.slug})
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_get_comment(
+        self,
+    ) -> None:
+        response = self.client.get(
+            reverse(
+                "comment-detail",
+                kwargs={
+                    "slug": self.article.slug,
+                    "lookup_id": self.comment.lookup_id,
+                },
+            ),
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_comment_without_authentication(
+        self,
+    ) -> None:
+        response = self.client.get(
+            reverse(
+                "comment-detail",
+                kwargs={
+                    "slug": self.article.slug,
+                    "lookup_id": self.comment.lookup_id,
+                },
+            ),
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_comment_with_invalid_lookup_id(
+        self,
+    ) -> None:
+        with self.assertRaises(ValidationError):
+            self.client.get(
+                reverse(
+                    "comment-detail",
+                    kwargs={
+                        "slug": self.article.slug,
+                        "lookup_id": "invalid-id",
+                    },
+                ),
+            )
+
+    def test_update_comment(
+        self,
+    ) -> None:
+        data = {"body": fake.text()}
+        response = self.client.put(
+            reverse(
+                "comment-detail",
+                kwargs={
+                    "slug": self.article.slug,
+                    "lookup_id": self.comment.lookup_id,
+                },
+            ),
+            data=data,
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
Assert that it is updated that the body has changed
Checkout the difference between put and patch